src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,512 @@
+        match self {
+            simple_extensions::ArgumentsItem::Variant0 {
+                name,
+                description,
+                options,
+            } => EnumArgument::parse(name, description, options).map(Into::into),
+
+            simple_extensions::ArgumentsItem::Variant1 {
+                name,
+                description,
+                value,
+                constant,
+            } => ValueArgument::parse(ctx, name, description, value, constant).map(Into::into),
+
+            simple_extensions::ArgumentsItem::Variant2 {
+                name,
+                description,
+                type_,
+            } => TypeArgument::parse(ctx, name, description, type_).map(Into::into),
+        }
+    }
+}
+
+impl From<ArgumentsItem> for simple_extensions::ArgumentsItem {
+    fn from(value: ArgumentsItem) -> Self {
+        match value {
+            ArgumentsItem::EnumArgument(EnumArgument {
+                name,
+                description,
+                options,
+            }) => simple_extensions::ArgumentsItem::Variant0 {
+                name: Some(name),
+                description: Some(description),
+                options,
+            },
+
+            ArgumentsItem::ValueArgument(ValueArgument {
+                name,
+                description,
+                value,
+                constant,
+            }) => simple_extensions::ArgumentsItem::Variant1 {
+                name: Some(name),
+                description: Some(description),
+                value,
+                constant: Some(constant),
+            },
+
+            ArgumentsItem::TypeArgument(TypeArgument {
+                name,
+                description,
+                type_,
+            }) => simple_extensions::ArgumentsItem::Variant2 {
+                name: Some(name),
+                description: Some(description),
+                type_,
+            },
+        }
+    }
+}
+
+/// Parse errors for [simple_extensions::ArgumentsItem].
+#[derive(Debug, Error, PartialEq)]
+pub enum ArgumentsItemError {
+    /// Empty enumeration option.
+    #[error("empty enumeration option")]
+    EmptyEnumOption,
+
+    /// Empty optional field.
+    #[error("the optional field `{0}` is empty and should be removed")]
+    EmptyOptionalField(String),
+}
+
+/// Arguments that support a fixed set of declared values as constant arguments.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
+
+    /// List of valid string options for this argument.
+    options: Vec<String>,
+}
+
+impl EnumArgument {
+    fn parse(
+        name: Option<String>,
+        description: Option<String>,
+        options: Vec<String>,
+    ) -> Result<EnumArgument, ArgumentsItemError> {
+        if options.iter().any(String::is_empty) {
```

Comment:
I didn't found this constraint in the doc. If you are sure about this, I will fix it.

Reply:
I'm not sure about this. The schema isn't strict about it, but I wonder if it makes sense to have enum args that have no "option"? I check the extensions in https://github.com/substrait-io/substrait/tree/main/extensions, and all enum args have non-empty options e.g. https://github.com/substrait-io/substrait/blob/de4fcbc066315ca874ca163275affdc4156e570e/extensions/functions_datetime.yaml#L71-L72. Maybe we should propose a change to the schema to require at least one option?

Reply:
After looking through the doc again, I agree with you that `options` should not be empty, otherwise, this argument serves no purpose.
Also in `substrait-java`([link](https://github.com/substrait-io/substrait-java/blob/79decd20e85d6a1a5623890042ebcf1415cf784a/core/src/main/java/io/substrait/expression/EnumArg.java#L25)), it implicitly requires `options` is not empty.